shop/src/main/java/ro/msg/learning/shop/service/OrderService.java

```diff
@@ -0,0 +1,107 @@
+package ro.msg.learning.shop.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.dto.OrderDTO;
+import ro.msg.learning.shop.dto.OrderDetailDTO;
+import ro.msg.learning.shop.dto.ProductDTO;
+import ro.msg.learning.shop.dto.StockDTO;
+import ro.msg.learning.shop.entity.*;
+import ro.msg.learning.shop.exception.ProductNotFoundException;
+import ro.msg.learning.shop.exception.ProductsCantBeShipped;
+import ro.msg.learning.shop.mapper.OrderDetailMapper;
+import ro.msg.learning.shop.mapper.OrderMapper;
+import ro.msg.learning.shop.mapper.ProductMapper;
+import ro.msg.learning.shop.repo.*;
+import ro.msg.learning.shop.config.ConfigureStrategy;
+import ro.msg.learning.shop.utils.LocationFormat;
+
+import java.time.LocalDateTime;
+import java.util.*;
+
+
+@Service
+@RequiredArgsConstructor
+public class OrderService {
+    private final OrderRepo orderRepository;
+    private final OrderMapper orderMapper;
+    private final OrderDetailRepo orderDetailsRepository;
+    private final OrderDetailMapper orderDetailMapper;
+    private final ProductRepo productRepository;
+    private final ProductMapper productMapper;
+    private final CustomerRepo customerRepository;
+    private final ConfigureStrategy strategyConfiguration;
+    private final LocationRepo locationRepository;
+
+
+    public OrderDTO createOrder(OrderDTO orderDTO) {
+        OrderDTO newOrder;
+        LocalDateTime localDateTime = LocalDateTime.now();
+        List<OrderDetail> result = registerOrderedProducts(orderDTO.getOrderedProducts(), false);
+        LocationFormat deliveryLocation = new LocationFormat(null, orderDTO.getAddress().getCity(), orderDTO.getAddress().getCountry());
+        List<StockDTO> strategyResult = strategyConfiguration.decideStrategy().implementStrategy(orderDetailMapper.orderDetailListToOrderDetailDTOList(result), deliveryLocation);
+        Set<Location> locationsForShippingProducts = getShippingLocations(strategyResult);
+        if (!strategyResult.isEmpty()) {
+            Optional<Customer> client = customerRepository.findById(orderDTO.getUserID());
```

Comment:
this method not only registers the order, but also looks for the customer. It could be splitted up into more methods to better reflect their purpose. Why? -> Separation of concerns.

The best rule of thumb to keep in mind: Controllers communicate with Services. Services communicate with their designated repo (this case only with OrderRepository) and with other services. Refactor this class in order to reflect this arhitectural design.